### ISSUE TITLE
fix(examples): Replace cross_entropy_loss with cross_entropy in training scripts

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2242]
## Problem
Training scripts import `cross_entropy_loss` and `cross_entropy_loss_backward` which do NOT exist. The actual functions in `shared/core/loss.mojo` are named `cross_entropy` and `cross_entropy_backward`.

This is a breaking API mismatch that prevents examples from compiling.

## Files to Modify
- `examples/alexnet-cifar10/train.mojo` (lines 27, 138, 143)
- `examples/vgg16-cifar10/train.mojo` (lines 27, 157, 162)

## Changes Required
```mojo
# FROM:
from shared.core.loss import cross_entropy_loss, cross_entropy_loss_backward
var loss = cross_entropy_loss(logits, labels)
var grad_logits = cross_entropy_loss_backward(logits, labels)

# TO:
from shared.core.loss import cross_entropy, cross_entropy_backward
var loss = cross_entropy(logits, labels)
var grad_logits = cross_entropy_backward(grad_output, logits, labels)
```

## Effort Estimate
1-2 hours

## Dependencies
None - this should be fixed immediately as it blocks examples from running.
